sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Testing/LiveResourceManager.cs

```diff
@@ -30,241 +30,281 @@ public sealed class LiveResourceManager
-                    return IsRetriableStatus(clEx.Response.StatusCode);
+                var groupData = new EventHubsConsumerGroupData();
 
-                case TimeoutException _:
-                case TaskCanceledException _:
-                case OperationCanceledException _:
-                case HttpRequestException _:
-                case WebException _:
-                case SocketException _:
-                case IOException _:
-                    return true;
+                await Task.WhenAll
+                (
+                    groups.Select(groupName =>
+                    {
+                        return consumerGroupCollection.CreateOrUpdateAsync(WaitUntil.Completed, groupName, groupData);
+                    })
+                ).ConfigureAwait(false);
+            }
 
-                default:
-                    return false;
-            };
+            return eventHub.Value;
         }
 
         /// <summary>
-        ///   Calculates the retry delay to use for management-related operations.
+        ///   Deletes an Event Hub under the active namespace used for the test run.
         /// </summary>
         ///
-        /// <param name="attempt">The current attempt number.</param>
-        /// <param name="exponentialBackoffSeconds">The exponential back-off amount,, in seconds.</param>
-        /// <param name="baseJitterSeconds">The amount of base jitter to include, in seconds.</param>
+        /// <param name="eventHubName">The name of the Event Hub to delete.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
         ///
-        /// <returns>The interval to wait before retrying the attempted operation.</returns>
-        ///
-        private static TimeSpan CalculateRetryDelay(int attempt,
-                                                    double exponentialBackoffSeconds,
-                                                    double baseJitterSeconds) =>
-            TimeSpan.FromSeconds((Math.Pow(2, attempt) * exponentialBackoffSeconds) + (RandomNumberGenerator.Value.NextDouble() * baseJitterSeconds));
+        public async Task DeleteEventHubAsync(string eventHubName,
+                                              CancellationToken cancellationToken = default) =>
+            await ArmClient.GetEventHubResource(GetEventHubResourceIdentifier(eventHubName)).DeleteAsync(WaitUntil.Completed, cancellationToken).ConfigureAwait(false);
 
         /// <summary>
-        ///   An internal type for tracking the management access token and
-        ///   its associated expiration.
+        ///   A retry policy that extends the common retry scenarios to include
+        ///   additional response types specific to Event Hubs management operations.
         /// </summary>
         ///
-        private class ManagementToken
+        /// <seealso cref="Azure.Core.Pipeline.RetryPolicy" />
+        ///
+        private class EventHubsManagementRetryPolicy : RetryPolicy
```

Comment:
Talked this over with Josh, and there's currently no way to override just the classifier used by the default retry policy.   He was kind enough to open #37499 to track.

In the meantime, I went with a custom policy that does just a in-place classifier swap in the scope of the retry policy only.  To be transparent, I wasn't sure if there was a downside to just swapping it out and leaving it.    Happy to remove the capture/restore for the original policy if it serves no purpose.

Reply:
The classifier would only be used in the retry policy so I think you are correct that restoring doesn't serve a purpose. Another options would be to just implement the logic in your override of ShouldRetry instead of calling base. That way you wouldn't have to override the classifier on the response.

Reply:
I'm trying to take advantage of the base tracking the retries and applying the delay, so I'm going to avoid taking that over.   With your insight, I can simplify quite a bit to just set the classifier and pass the buck.